pymc3/tests/test_distributions_random.py

```diff
@@ -622,6 +627,24 @@ def ref_rand(size, tau):
 
         pymc3_random(pm.HalfNormal, {"tau": Rplus}, ref_rand=ref_rand)
 
+    def test_zerosum_normal():
+        def ref_rand(size, sigma):
+            zerosum_axes = ()
+            shape = ()
+            zerosum_axes = [a if a >= 0 else len(shape) + a for a in zerosum_axes]
+            _degrees_of_freedom = np.prod(
+                [s if axis not in zerosum_axes else s - 1 for axis, s in enumerate(shape)]
+            )
+            _full_size = np.prod(shape)
+            sigma_factor = np.sqrt(_full_size / _degrees_of_freedom)
+
+            samples = st.norm.rvs(loc=0, scale=sigma * sigma_factor, size=size)
+            for axis in zerosum_axes:
+                samples -= np.mean(samples, axis=axis, keepdims=True)
+            return samples
+
+        pymc3_random(pm.HalfNormal, {"sigma": Rplus}, ref_rand=ref_rand)
```

Comment:
Shouldn't this be `pm.ZeroSumNormal`?
Maybe it would be better to draw the samples using `stats.multivariate_normal(cov=sigma ** 2 * (np.eye(n) - np.ones(n)/n))`. That way we check for the correct covariance.

Reply:
Thanks for the tip, did just that.